packages/flutter/lib/src/cupertino/text_form_field_row.dart

```diff
@@ -0,0 +1,391 @@
+/// ```
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return CupertinoPageScaffold(
+///     child: Center(
+///       child: Form(
+///         autovalidateMode: AutovalidateMode.always,
+///         onChanged: () {
+///           Form.of(primaryFocus.context).save();
+///         },
+///         child: CupertinoFormSection.padded(
+///           header: Text('SECTION 1'),
+///           children: List<Widget>.generate(5, (int index) {
+///             return CupertinoTextFormFieldRow(
+///               prefix: Text('Enter text'),
+///               placeholder: 'Enter text',
+///               validator: (value) {
+///                 if (value.isEmpty) {
+///                   return 'Please enter a value';
+///                 }
+///                 return null;
+///               },
+///             );
+///          }),
+///         ),
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+class CupertinoTextFormFieldRow extends FormField<String> {
+  /// Creates a [CupertinoFormRow] containing a [FormField] that wraps
+  /// a [CupertinoTextField].
+  ///
+  /// When a [controller] is specified, [initialValue] must be null (the
+  /// default). If [controller] is null, then a [TextEditingController]
+  /// will be constructed automatically and its `text` will be initialized
+  /// to [initialValue] or the empty string.
+  ///
+  /// The [prefix] parameter is displayed at the start of the row. Standard iOS
+  /// guidelines encourage passing a [Text] widget to [prefix] to detail the
+  /// nature of the input.
+  ///
+  /// The [margins] parameter is used to pad the contents of the row. It is
+  /// directly passed to [CupertinoFormRow]. If the [margins]
+  /// parameter is null, [CupertinoFormRow] constructs its own default
+  /// margins (which are the standard form row margins in iOS.) If no edge
+  /// insets are intended, explicitly pass [EdgeInsets.zero] to [margins].
+  ///
+  /// For documentation about the various parameters, see the
+  /// [CupertinoTextField] class and [new CupertinoTextField.borderless],
+  /// the constructor.
+  CupertinoTextFormFieldRow({
+    Key? key,
+    this.prefix,
+    this.margins,
+    this.controller,
+    String? initialValue,
+    FocusNode? focusNode,
+    BoxDecoration? decoration,
+    TextInputType? keyboardType,
+    TextCapitalization textCapitalization = TextCapitalization.none,
+    TextInputAction? textInputAction,
+    TextStyle? style,
+    StrutStyle? strutStyle,
+    TextAlign textAlign = TextAlign.start,
+    TextAlignVertical? textAlignVertical,
+    bool autofocus = false,
+    bool readOnly = false,
+    ToolbarOptions? toolbarOptions,
+    bool? showCursor,
+    String obscuringCharacter = '•',
+    bool obscureText = false,
+    bool autocorrect = true,
+    SmartDashesType? smartDashesType,
+    SmartQuotesType? smartQuotesType,
+    bool enableSuggestions = true,
+    bool maxLengthEnforced = true,
```

Comment:
nit: this currently doesn't do anything iirc, and I'd expect it to be removed in the future. Not sure if we want to add it in a new API.